packages/main/src/components/AnalyticalTable/index.tsx

```diff
@@ -655,6 +656,105 @@ const AnalyticalTable = forwardRef<AnalyticalTableDomRef, AnalyticalTablePropTyp
     );
   };
 
+  const handleOnAutoResize = (e, accessor) => {
+    // Helpers
+    interface canvasHolderProps {
+      canvas?: HTMLCanvasElement;
+    }
+    const canvasHolder: canvasHolderProps = { canvas: undefined };
+    // Text Width Analysis
+    function getTextWidth(text: string, font: string) {
+      // Reusing the canvas is more efficient
+      const canvas = canvasHolder.canvas || (canvasHolder.canvas = document.createElement('canvas'));
+      const context = canvas.getContext('2d');
+      context.font = font;
+      const metrics = context.measureText(text);
+      return metrics.width;
+    }
+
+    function getCssStyle(element: Element, prop: string) {
+      return window.getComputedStyle(element, null).getPropertyValue(prop);
+    }
+
+    function getCanvasFont(el: Element = document.body) {
+      const fontWeight = getCssStyle(el, 'font-weight') || 'normal';
+      const fontSize = getCssStyle(el, 'font-size') || '12px';
+      const fontFamily = getCssStyle(el, 'font-family') || 'Times New Roman';
+
+      return `${fontWeight} ${fontSize} ${fontFamily}`;
+    }
+
+    const findWidth = (text: string, el: Element) => {
+      let font: string | undefined;
+      if (!font) font = getCanvasFont(el);
+      return getTextWidth(text, font);
+    };
+    // End Helpers
+    let largest = 0;
+    // Currently Including Overscan
+    const items = rowVirtualizer.getVirtualItems();
+    const [start, end] = [items[0].index, items[items.length - 1].index];
+
+    for (let i = start; i < end; i++) {
+      // Use the classname for the span where the text lives AnalyticalTable.module.css.js
+      const collection = document.getElementsByClassName(clsx(classNames.tableText));
```

Comment:
Searching for a single element should be sufficient here, also I don't think that you need to do that inside the loop, or am I missing something here?
In general, I believe it's enough to retrieve the canvas font once when a user double-clicks the resizer. Given that only standard cell content (text) is supported, with no custom components, the values should be the same for each cell.